src/Microsoft.AspNetCore.Mvc.Core/Internal/DefaultCollectionValidationStrategy.cs

```diff
@@ -45,6 +46,7 @@ public class DefaultCollectionValidationStrategy : IValidationStrategy
         /// Gets an instance of <see cref="DefaultCollectionValidationStrategy"/>.
         /// </summary>
         public static readonly IValidationStrategy Instance = new DefaultCollectionValidationStrategy();
+        private static readonly Dictionary<Type, MethodInfo> _genericGetEnumeratorCache = new Dictionary<Type, MethodInfo>();
```

Comment:
@dougbu I made the cache static as it's used from `GetEnumeratorForElementType()` which is itself static. I guess I could access it through the singleton, with something along the lines of:
```C#
public static IEnumerator GetEnumeratorForElementType(ModelMetadata metadata, object model)
{
    MethodInfo getEnumeratorMethod;
    var instance = (DefaultCollectionValidationStrategy)Instance; 
    if (!instance._genericGetEnumeratorCache.TryGetValue(metadata.ElementType, out getEnumeratorMethod))
    {
        getEnumeratorMethod = _getEnumerator.MakeGenericMethod(metadata.ElementType);
        instance._genericGetEnumeratorCache.Add(metadata.ElementType, getEnumeratorMethod);
    }
    return (IEnumerator)getEnumeratorMethod.Invoke(null, new object[] { model });
}
```

Is that a reasonable approach?

I agree on the `ConcurrentDictionary` suggestion, will send that change soon.